packages/core/upload/admin/src/components/TableList/TableRows.js

```diff
@@ -9,32 +9,44 @@ import { Typography } from '@strapi/design-system/Typography';
 import Pencil from '@strapi/icons/Pencil';
 
 import { PreviewCell } from './PreviewCell';
-import { AssetDefinition } from '../../constants';
+import { AssetDefinition, FolderDefinition } from '../../constants';
 import { formatBytes, getTrad } from '../../utils';
 
-export const TableRows = ({ assets, onEditAsset, onSelectAsset, selectedAssets }) => {
+export const TableRows = ({ onEditAsset, onEditFolder, onSelectOne, rows, selected }) => {
   const { formatDate, formatMessage } = useIntl();
 
   return (
     <Tbody>
-      {assets.map((asset) => {
-        const { alternativeText, id, name, ext, size, createdAt, updatedAt, url, mime, formats } =
-          asset;
+      {rows.map((element) => {
+        const {
+          alternativeText,
+          id,
+          name,
+          ext,
+          size,
+          createdAt,
+          updatedAt,
+          url,
+          mime,
+          formats,
+          type,
+        } = element || {};
```

Comment:
I'm wondering if it would be possible to find something a bit more flexible here. One thing could be a type-mapping:

```
const typeMap = {
  'date': ['created_at'],
}
```

Later on you could then loop over every attribute of `element`, find the appropriate type and render a column. Whenever a user adds an attribute to an asset (which is possible I think) it could be rendered in the table.

Reply:
that sounds nice 🤔
only thing is we still have to render elements in the same table which do not all have the same keys: 

- folder only have `type`, `id`, `name`, `createdAt`, `updatedAt`, but we still need to render an `ext` and `size` placeholder information
- I need to extract `alternativeText`, `ext`, `size`, `url`, `mime`, `formats` for assets which folder don't have

Reply:
I don't think that would be a problem and depends on how you approach it. You can create two maps (folders/ assets) or merge all in one. You could also aggregate all attributes upfront and merge folder + asset attributes. Keep in mind that at some point which columns to display might become an option (similar to the CM) ...

Reply:
I think it looks good to be honest, very clear, I wrote a comment on the tests stuff though 😄